cpp/inc/bond/protocol/simple_json_reader.h

```diff
@@ -39,11 +39,12 @@ class SimpleJsonReader
         : _input(input),
           _stream(_input),
           _document(boost::make_shared<rapidjson::Document>()),
-          _value(NULL)
+          _value(nullptr)
     {}
 
     SimpleJsonReader(const SimpleJsonReader& that, const Field& value)
-        : _stream(that._stream),
+        : _input(that._input),
+          _stream(that._stream),
```

Comment:
So in this case `_stream.input` will point to a buffer the lifetime of which we do not control?

Reply:
The caller is already required to keep the input buffer alive as long as the stream is alive.
Simple JSON doesn't give you back a reference to a blob: it has to copy and decode the bytes, so the lifetime issues shouldn't be a problem there.

Perhaps Simple JSON just shouldn't expose its underlying buffer at all... Thoughts?

Reply:
Let me try switching this to have a "parent" SimpleJsonReader instead of copying the input and stream.

Reply:
Yeah, that sounds better.